### PR TITLE
cmd: add benchmark tool 'benchdb'

### PR DESCRIPTION
**Pull request by coocood**
'benchdb' is used to make testing basic table performance easier.
Also move 'benchkv' to 'cmd' folder.

### CODE REVIEW

**Comment by coocood** [comment]
Some documentations need to be added.

File: cmd/benchdb/main.go [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+		"update-range:5000_5100:1000",
+		"select:0_10000:10",
+		"gc",
+		"select:0_10000:10",
+	}, ","), "work-flow to run")
+)
+
+var blobString string
+
+func main() {
```
Comment (shenli) [inline cmd/benchdb/main.go]: Add a short description about this bench tool.

File: cmd/benchdb/main.go [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+	}
+	cLog("GC finished, duration ", time.Since(start))
+	if done != nil {
+		done <- true
+	}
+}
+
+func cLogf(format string, args ...interface{}) {
+	str := fmt.Sprintf(format, args...)
+	fmt.Println("\033[0;32m" + str + "\033[0m\n")
```
Comment (coocood) [inline cmd/benchdb/main.go]: It's color code.

File: cmd/benchdb/main.go (hunk 1 above)
Comment (coocood) [inline cmd/benchdb/main.go]: Added README file.

**Comment by coocood** [comment]
@shenli @hanfei1991 PTAL

**Comment by zimulala** [comment]
LGTM

File: cmd/benchdb/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+## BenchDB
+
+benchdb is a command line tool to test the performance of TiDB.
+
+### QuickStart
```
Comment (queenypingcap) [inline cmd/benchdb/README.md]: Quick Start

File: cmd/benchdb/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+Make sure you have started PD and TiKV, then run:
+
+```
+    ./benchdb -run="create,truncate,insert:0_10000,update-random:0_10000:100000,select:0_10000:10"
```
Comment (queenypingcap) [inline cmd/benchdb/README.md]: Why the extra white space at the beginning?

File: cmd/benchdb/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+```
+
+
+### Arguments
+
+#### `run`
+The `run` argument defines the workflow of the test, you can define
```
Comment (queenypingcap) [inline cmd/benchdb/README.md]: , you -> . You

File: cmd/benchdb/README.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+multiple jobs, separated by `,`, the jobs are executed sequentially.
```
Comment (queenypingcap) [inline cmd/benchdb/README.md]: , the -> . The

File: cmd/benchdb/README.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+ 
+* `create` creates a table, currently it's just a typical simple table, with a few columns.
```
Comment (queenypingcap) [inline cmd/benchdb/README.md]: In "currently it's just a typical simple table ", what does it refer to?

File: cmd/benchdb/README.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+* `truncate` truncates the table.
+
+* `insert:xxx_yyy` inserts rows starts with ID in `[xxx, yyy)`.
```
Comment (queenypingcap) [inline cmd/benchdb/README.md]: starts -> starting

File: cmd/benchdb/README.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+ e.g. `insert:0_10000` inserts 10000 rows with id in range  `[0, 9999)`.
```
Comment (queenypingcap) [inline cmd/benchdb/README.md]: id -> ID

File: cmd/benchdb/README.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+   
+* `update-random:xxx_yyy:zzz` updates a row with a random ID in range `[xxx, yyy)`, do `zzz` times.
+ e.g. `update-random:100_200:50` updates 50 random rows with ID in range `[100, 200)`.
+ 
+* `update-range:xxx_yyy:zzz` update a range of rows with ID in range `[xxx, yyy)`, do `zzz` times.
```
Comment (queenypingcap) [inline cmd/benchdb/README.md]: -> updates a range of rows with ID in range `[xxx, yyy)` for `zzz` times.

File: cmd/benchdb/README.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+* `insert:xxx_yyy` inserts rows starts with ID in `[xxx, yyy)`.
+ e.g. `insert:0_10000` inserts 10000 rows with id in range  `[0, 9999)`.
+   
+* `update-random:xxx_yyy:zzz` updates a row with a random ID in range `[xxx, yyy)`, do `zzz` times.
+ e.g. `update-random:100_200:50` updates 50 random rows with ID in range `[100, 200)`.
+ 
+* `update-range:xxx_yyy:zzz` update a range of rows with ID in range `[xxx, yyy)`, do `zzz` times. 
+  
+* `select:xxx_yyy:zzz` select rows with ID range in `[xxx, yyy)`, do `zzz` times.
```
Comment (queenypingcap) [inline cmd/benchdb/README.md]: selects rows with ID range in `[xxx, yyy)` for `zzz` times.

File: cmd/benchdb/README.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+
+* `gc` does a manually triggered GC, so we can compare the performance before and after GC. 
+ 
+ The output shows execution time.
```
Comment (queenypingcap) [inline cmd/benchdb/README.md]: execution time -> the execution time

File: cmd/benchdb/README.md [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+ 
+#### `table`
+
+Name of the table, so we can create many tables for different tests without the need to clean up.
```
Comment (queenypingcap) [inline cmd/benchdb/README.md]: Name of the table -> The name of the table

File: cmd/benchdb/README.md [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+Default is `bench_db`.
+  
+#### `blob`
+
+blob column size in bytes, so we can test performance for different row size.
```
Comment (queenypingcap) [inline cmd/benchdb/README.md]: blob column size -> The blob column size

File: cmd/benchdb/README.md [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+Default is `1000`.
+
+#### `batch`
+
+batch number of statements in a transaction, used for insert and update-random only, to speed up the test workflow.
```
Comment (queenypingcap) [inline cmd/benchdb/README.md]: -> The batch number of the statements in a transaction which is used

File: cmd/benchdb/README.md [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+Default is `100`.
+
+#### `addr`
+
+PD address. Default is `127.0.0.1:2379`.
```
Comment (queenypingcap) [inline cmd/benchdb/README.md]: PD address -> The PD address

File: cmd/benchdb/README.md [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+
+### `L`
+
+log level. Default is `warn`.
```
Comment (queenypingcap) [inline cmd/benchdb/README.md]: log level -> The log level.

Comment (queenypingcap) [inline cmd/benchdb/README.md]: Can we add an example to show how the test result looks like?

File: cmd/benchdb/README.md (hunk 7 above)
Comment (coocood) [inline cmd/benchdb/README.md]: The table definition may change, so I don't want to make it specific in README.

File: cmd/benchdb/README.md (hunk 17 above)
Comment (coocood) [inline cmd/benchdb/README.md]: The result output may change frequently.
If added it in README, we need to update README whenever we change the output.

**Comment by queenypingcap** [comment]
LGTM

**Pull request merged**